### PR TITLE
Feature/422 script action

### PR DESCRIPTION
**Pull request by zzeekk**
### What changes are included in the pull request?
see #422
and there is a bigger refactoring of SubFeed handling to reduce code duplications and prepare for new engines.

### Why are the changes needed?
see #422 + prepare for new engines, e.g. #376

### CODE REVIEW

File: sdl-core/src/main/scala/io/smartdatalake/workflow/action/ActionHelper.scala [hunk 1]
```diff
@@ -127,6 +127,7 @@ private[smartdatalake] object ActionHelper extends SmartDataLakeLogger {
     invalidCharacters.replaceAllIn(str, "_")
   }
 
+  // TODO: remove
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/workflow/action/ActionHelper.scala]: Are these deprecated then? Or can they be removed now?

File: sdl-core/src/main/scala/io/smartdatalake/workflow/action/FileOneToOneActionImpl.scala [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+import io.smartdatalake.config.ConfigurationException
+import io.smartdatalake.definitions.{Environment, SDLSaveMode}
+import io.smartdatalake.util.hdfs.PartitionValues
+import io.smartdatalake.util.misc.PerformanceUtils
+import io.smartdatalake.workflow._
+import io.smartdatalake.workflow.dataobject._
+import org.apache.spark.sql.SparkSession
+
+abstract class FileOneToOneActionImpl extends ActionSubFeedsImpl[FileSubFeed] {
```
Comment (pgruetter) [inline sdl-core/src/main/scala/io/smartdatalake/workflow/action/FileOneToOneActionImpl.scala:32]: There's a few classes without a doc. Can you please add a short description ?

**Review by pgruetter — APPROVED** [review]
See comments, other than that OK for me.

File: sdl-core/src/main/scala/io/smartdatalake/workflow/action/ActionHelper.scala (hunk 1 above)
Comment (zzeekk) [reply sdl-core/src/main/scala/io/smartdatalake/workflow/action/ActionHelper.scala]: yes, can be removed, thanks!

File: sdl-core/src/main/scala/io/smartdatalake/workflow/action/FileOneToOneActionImpl.scala (hunk 2 above)
Comment (zzeekk) [reply sdl-core/src/main/scala/io/smartdatalake/workflow/action/FileOneToOneActionImpl.scala:32]: ok, done

**Comment by pgruetter** [comment]
@zzeekk https://github.com/smart-data-lake/smart-data-lake/pull/425/commits/81a65599db1ebf728b464f52c8fc38633f2ec3e8 fixed the build problem for me, thanks!

**Pull request merged**